### ISSUE TITLE
Key Signature Changes shown incorrectly in Braille

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#34354]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

Follow the attached score to see the output:

Bar 1 starts in Eb Major – This is shown correctly in Braille after treble clef and before time signature before bar 1 as expected.
Bar 5 start of second system – Again shown correctly in Braille with clef and key as in print. 
Bar 6 key change to C Major – Strangely the Braille shows a treble clef before this bar and does not show the 3 naturals.
Bar 9 start of second system – Shows the clef and no key signature, correct for C Major.
Bar 10 – Change to Eb Major – Shown correctly in Braille but again with an added treble clef that I assume is not there in print.
Bar 11 Change to Bb Major – Again the 2 flats are shown correctly but with added treble clef before.
Bar 13 start of system 3 – Correct Braille output of clef and key signature as in print.
Bar 13 – system break at end of bar – The warning key signature change to Eb is not shown in Braille.
Bar 14 change to Eb Major is displayed correctly in Braille as it is the start of the new system.

The clef should not be shown in Braille when the key signature change is mid line.  This is important as currently the only way for a blind user to determine layout in Braille is because the line starts with a clef and key signature so this output is confusing.

If the key Change is to C Major the naturals need to be shown in Braille as currently if the clef was removed there is nothing there at all to show this key change.

Warning changes on ends of lines aren’t something generally used in Braille, but as the Braille output in MuseScore is generally showing the print format they should probably be included also.  But at the very least the naturals to signify a change to C Major would be good included as it is easy to not spot the next line simply not having a key signature when others have.

### Supporting files, videos and screenshots

[key signature Braille issues.zip](https://github.com/user-attachments/files/30395804/key.signature.Braille.issues.zip)

### In which versions of MuseScore Studio is this issue present?

4.7.4

### Regression

I was unable to check

### Operating system

Windows 11 with NVDA

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible
